### PR TITLE
many: don't concatenate non-constant format strings

### PR DESCRIPTION
**Pull request by mardy**
We should be careful not to concatenate variable strings into the first argument of Sprintf/Printf/Errorf: if these variable strings end up containing a percent character, it will break the way that the printfs arguments are interpreted. Luckily, golang is smarter than C and is able to detect mismatches between the number of '%' and the number of arguments, but it still can lead to unexpected results:

    $ sudo snap set core refresh.rate-limit='%[2]#v'
    error: cannot perform the following tasks:
    - Run configure hook of "core" snap (run hook "configure": cannot parse "%!#(BADINDEX)v": no numerical prefix)

also:

    $ sudo snap set core refresh.rate-limit='%#v'
    error: cannot perform the following tasks:
    - Run configure hook of "core" snap (run hook "configure": cannot parse "&errors.errorString{s:"no numerical prefix"}": %!s(MISSING))

Moreover, it appears that all the occurrences of such pattern in our code are situated either on unprivileged processes (like the `snap` client), or, when in snapd, can only be triggered by the root user (notice the `sudo` in the commands above).

Nevertheless, let's be defensive and fix these.

There are also other occurrences of concatenations in formatting strings, but those are only constants so they don't pose a problem. But to avoid the risk of these strings getting updated in the future with a mutable version, let's explicitly mark these format prefixes as `const`.

### CODE REVIEW

**Review by MiguelPires — COMMENTED** [review]
There are 2 mismatches of format specifiers and arguments

File: daemon/api_users.go [hunk 1]
```diff
@@ -501,26 +501,26 @@ func getUserDetailsFromAssertion(st *state.State, modelAs *asserts.Model, serial
 	// check that the signer of the assertion is one of the accepted ones
 	sysUserAuths := modelAs.SystemUserAuthority()
 	if len(sysUserAuths) > 0 && !strutil.ListContains(sysUserAuths, su.AuthorityID()) {
-		return "", nil, fmt.Errorf(errorPrefix+"%q not in accepted authorities %q", email, su.AuthorityID(), sysUserAuths)
+		return "", nil, fmt.Errorf("%s%q not in accepted authorities %q", errorPrefix, email, su.AuthorityID(), sysUserAuths)
```
[the remark this answers is not in the document]
Comment (mardy) [reply daemon/api_users.go:504]: Wow, good catches!!!

**Review by valentindavid — APPROVED** [review]
Good!

**Review by MiguelPires — APPROVED** [review]
LGTM, thanks

**Pull request merged**